File: implementations/java-springboot/examples/src/main/java/com/example/accountservice/account/application/command/DepositByPaymentCommand.java
package com.example.accountservice.account.application.command;

// Payment BC's paymentId (payment-cancellation compensating credit) or refundId (refund-approval
// credit). Used as the key for the idempotency check (Level 2 Ledger).
public record DepositByPaymentCommand(String accountId, long amount, String referenceId) {}

File: implementations/kotlin-springboot/examples/src/main/kotlin/com/example/accountservice/account/application/command/DepositByPaymentCommand.kt
package com.example.accountservice.account.application.command

data class DepositByPaymentCommand(
    val accountId: String,
    val amount: Long,
    // The Payment BC's paymentId (cancellation compensation credit) or refundId (refund-approval
    // credit). Used as the key for idempotency checks (Level 2 Ledger).
    val referenceId: String,
)
